common/djangoapps/third_party_auth/api/permissions.py

```diff
@@ -14,39 +13,12 @@
 )
 from rest_condition import C
 from rest_framework.permissions import BasePermission
-from third_party_auth.models import ProviderApiPermissions
 
 from openedx.core.lib.api.permissions import ApiKeyHeaderPermission
 
 log = logging.getLogger(__name__)
 
 
-class ThirdPartyAuthProviderApiPermission(BasePermission):
-    """
-    Allow someone to access the view if they have valid OAuth client credential.
-
-    Deprecated: Only works for DOP oauth applications. To be removed as part of DOPrecation.
-
-    """
-    def has_permission(self, request, view):
-        """
-        Check if the OAuth client associated with auth token in current request has permission to access
-        the information for provider
-        """
-        provider_id = view.kwargs.get('provider_id')
-        if not request.auth or not provider_id:
-            # doesn't have access token or no provider_id specified
-            return False
-
-        try:
-            ProviderApiPermissions.objects.get(client__pk=request.auth.client_id, provider_id=provider_id)
-        except ProviderApiPermissions.DoesNotExist:
-            return False
-
-        set_custom_metric('deprecated_ThirdPartyAuthProviderApiPermission', True)
```

Comment:
Do you mind doing a quick review of this in NewRelic?

Reply:
This was not null only once in new relic(on feb 24, 2020 and the user was UBC)